### PR TITLE
Specify target type for Network.Parse

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: NBitcoin/Network.cs [hunk 1]
```diff
@@ -2413,16 +2413,20 @@ public BitcoinScriptAddress CreateBitcoinScriptAddress(string base58)
...
 		}
 
-		public IBitcoinString Parse(string str)
+		public IBitcoinString Parse(string str, Type? targetType = null)
 		{
-			return Parse<IBitcoinString>(str, this);
+			return Parse<IBitcoinString>(str, this, targetType);
 		}
-		public T Parse<T>(string str) where T : IBitcoinString
+		public T Parse<T>(string str, Type? targetType = null) where T : IBitcoinString
```
Comment (NicolasDorier) [inline NBitcoin/Network.cs]: the `targetType` makes no sense here

Comment (NicolasDorier) [reply NBitcoin/Network.cs]: `Parse<T>` should call `Parse` not the reverse.

Comment (Kukks) [reply NBitcoin/Network.cs]: fixeed

**Comment by NicolasDorier** [comment]
tests not passing

File: NBitcoin/Network.cs [hunk 2]
```diff
@@ -2413,17 +2413,24 @@ public BitcoinScriptAddress CreateBitcoinScriptAddress(string base58)
...
-		public T Parse<T>(string str) where T : IBitcoinString
+		public IBitcoinString Parse(string str, Type targetType = null)
 		{
 			if (str == null)
 				throw new ArgumentNullException(nameof(str));
-			if (NetworkStringParser.TryParse<T>(str, this, out T o))
+			if (targetType == null)
+			{
+				targetType = typeof(IBitcoinString);
+			}else if (targetType.GetTypeInfo().IsAssignableFrom(typeof(IBitcoinString).GetTypeInfo()))
```
Comment (NicolasDorier) [inline NBitcoin/Network.cs]: you screwed up here

**Pull request merged**